linker/NetCore.props

```diff
@@ -1,13 +1,13 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <TargetFrameworks>netcoreapp2.0;net46</TargetFrameworks>
     <RuntimeFrameworkVersion>2.0.0-beta-001509-00</RuntimeFrameworkVersion>
     <DefineConstants>$(DefineConstants);NET_CORE</DefineConstants>
```

Comment:
Good question. I think the `NET_CORE` here is a bit misleading - it enables the behavior we want in illink but not monolinker. At the moment it just enables simplified assembly resolution logic and changes the usage message. Maybe in the future if we want different behavior on net46 compared to netcoreapp2.0, we should use different defines for the two purposes, but for now I think this is what we want.